modules/ui/combobox.js

```diff
@@ -386,7 +386,17 @@ export function uiCombobox(context, klass) {
                     return 'combobox-option ' + (d.klass || '');
                 })
                 .attr('title', function(d) { return d.title; })
-                .html(function(d) { return d.display || d.value; })
+                .html(function(d) {
+                    // d.display can be an object
+                    if ( typeof d.display === 'object' &&
+                         !Array.isArray(d.display) &&
+                         d.display !== null
```

Comment:
Does the Wikibase API potentially return either format? I thought it had already cut over to the new format irreversibly.

Reply:
True the Wikibase API returns `display` as an object, this property is newly introduced, but I think this part of the code is used by several input dropdown boxes handling other APIs as well to display dropdown values. 

In those other APIs, there is a field called `display` which is just a normal string.
Well, I just used console.log() to check `d.` Not sure if that's the right way.

Reply:
Ah, I missed that this is in the shared combobox code. Do you think it would be possible to modify the code specifically for the Wikidata field instead? That will avoid any potential impact on other fields that use comboboxes.

https://github.com/openstreetmap/iD/blob/7b0127f5e7b98214782574698c5fc8c970ed8caa/modules/ui/fields/wikidata.js#L357-L371

/ref https://github.com/openstreetmap/iD/issues/9067#issuecomment-1098414653

Reply:
Hey @1ec5,
Hmm. I think the above code `iD/modules/ui/fields/wikidata.js` > `Lines 357 to 371 in 7b0127f` only comes into play after an item is selected from the dropdown and not while showing the list of options in the dropdown (where our current problem is)
Additionally, it has a different data shape.
For example, `wikidataEntity` is
`
{
    "type": "item",
    "id": "Q23635",
    "labels": {
        "ms": {
            "language": "ms",
            "value": "Bermuda"
        },
        "en": {
            "language": "en",
            "value": "Bermuda"
        },
        "hi": {
            "language": "hi",
            "value": "बरमूडा"
        }
    },
    "descriptions": {...},
    "claims": {...},
    "sitelinks": {...}
}
`
`propKey `is `labels` & ultimately a string value in one of ['Bermuda', 'Bermuda', 'बरमूडा'], depending on the language, is returned.

Isn't this different for our context problem at hand here?
According to https://phabricator.wikimedia.org/T304878, the API changes `label` & `description` and not `labels` or `descriptions`
Or maybe I understood it wrong? Thoughts?

P.S. - Thanks for the comment.